extensions/funqy/funqy-knative-events/runtime/src/main/java/io/quarkus/funqy/knative/events/CloudEvent.java

```diff
@@ -1,19 +1,32 @@
 package io.quarkus.funqy.knative.events;
 
+import java.net.URI;
 import java.time.OffsetDateTime;
+import java.util.Iterator;
+import java.util.Map;
 
 /**
- * Cloud event. Represents only the headers. No data.
+ * CloudEvent.
  *
  */
-public interface CloudEvent {
+public interface CloudEvent<T> {
+
     String id();
 
     String specVersion();
 
-    String source();
+    URI source();
+
+    String type();
 
     String subject();
 
     OffsetDateTime time();
+
+    Iterator<Map.Entry<String, String>> extensions();
```

Comment:
why is this an integrator instead of just a map? just out of curiosity what is the benefit.

Reply:
I didn't want to allocate Map, but given there will be just few extensions it is probably premature optimization.

Reply:
+1 to using a map.  This is a case where user friendliness would be more important as I bet users would be interested in the specific value of an extension and having to go through an iterator is not user friendly.  

On a related note, @matejvasek There's a ton of optimizations that will need to be done all across the code base here.   Will probably attack them when I look at Qson integration.